File: src/Services/PromptBuilderService.php
<?php

namespace App\Services;

class PromptBuilderService
{
    /**
     * Build complete prompt for AI
     */
    public static function buildPrompt($documentText, $userQuestion)
    {
        $systemInstruction = "You are a document question-answering assistant.
Answer only from the provided document.
If the answer is not clearly stated in the document, say:
\"This is not clearly stated in the uploaded document.\"
Keep answers clear, concise, and professional.";

        $prompt = $systemInstruction . "\n\n";
        $prompt .= "DOCUMENT CONTENT:\n";
        $prompt .= "==================================================\n";
        $prompt .= $documentText . "\n";
        $prompt .= "==================================================\n\n";
        $prompt .= "USER QUESTION:\n";
        $prompt .= $userQuestion . "\n\n";
        $prompt .= "ANSWER:\n";

        return $prompt;
    }
}

File: services/prompt_builder_service.py
class PromptBuilderService:
    """Service for building prompts for AI"""

    @staticmethod
    def build_prompt(document_text, user_question):
        """Build complete prompt for AI"""
        system_instruction = """You are a document question-answering assistant.
Answer only from the provided document.
If the answer is not clearly stated in the document, say:
"This is not clearly stated in the uploaded document."
Keep answers clear, concise, and professional."""

        prompt = system_instruction + "\n\n"
        prompt += "DOCUMENT CONTENT:\n"
        prompt += "==================================================\n"
        prompt += document_text + "\n"
        prompt += "==================================================\n\n"
        prompt += "USER QUESTION:\n"
        prompt += user_question + "\n\n"
        prompt += "ANSWER:\n"

        return prompt
